src/io.rs

```diff
@@ -14,22 +18,36 @@ where
     F: Fn(&mut I, M) -> Result<M, String>,
 {
     return match File::open(&Path::new(path)) {
-        Ok(mut is) => match ::protobuf::parse_from_reader::<M>(&mut is) {
-            Ok(t) => callback(src, t),
-            Err(_) => Err(String::from("Failed to load file.")),
-        },
+        Ok(mut is) => {
+            match ::protobuf::parse_from_reader::<M>(&mut is) {
+                Ok(t) => callback(src, t),
+                Err(_) => Err(String::from("Failed to load file.")),
+            }
+        }
         Err(_) => Err(String::from("Failed to open file")),
     };
 }
 
+pub fn write_protobuf<I: Interfaceable>(
+    src: &mut I,
+    mut game: Game,
+) -> Result<Game, String> {
+
+    let mut file = save(src, &game);
+    let mut cos = CodedOutputStream::new(&mut file);
+    game.write_to(&mut cos);
+    cos.flush();
```

Comment:
warning: unused `std::result::Result` which must be used
  --> src/io.rs:38:5
   |
38 |     game.write_to(&mut cos);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_must_use)] on by default

adding .expect() may work

Reply:
game.write_to(&mut cos).expect("Something went wrong");

works